clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3269,6 +3279,51 @@ static void handleIntelReqdSubGroupSize(Sema &S, Decl *D,
   S.AddIntelReqdSubGroupSize(D, AL, E);
 }
 
+IntelNamedSubGroupSizeAttr *
+Sema::MergeIntelNamedSubGroupSizeAttr(Decl *D,
+                                      const IntelNamedSubGroupSizeAttr &A) {
+  if (checkAttrMutualExclusion<IntelNamedSubGroupSizeAttr>(*this, D, A))
+    return nullptr;
+  if (checkAttrMutualExclusion<SYCLSimdAttr>(*this, D, A))
+    return nullptr;
+
+  // Check to see if there's a duplicate attribute with different values
+  // already applied to the declaration.
+  if (const auto *DeclAttr = D->getAttr<IntelNamedSubGroupSizeAttr>()) {
+    if (DeclAttr->getType() != A.getType()) {
+      Diag(DeclAttr->getLoc(), diag::warn_duplicate_attribute) << &A;
+      Diag(A.getLoc(), diag::note_previous_attribute);
+      return nullptr;
+    }
+  }
```

Comment:
should we not drop duplicate attribute here?

Reply:
I think so. We do not want to duplicate the attribute here. @AaronBallman, any thought?

Reply:
Actually, doesn't the 'return nullptr' avoid adding the current one?

Reply:
It avoids adding the current one, but you're only avoiding the duplicate when the arguments don't match. If the arguments *do* match, we still don't need to add the duplicate attribute (the initial one suffices).

Reply:
return nullptr will be out side if (DeclAttr->getType() != A.getType()) { statement to avoid adding one.

Reply:
>It avoids adding the current one, but you're only avoiding the duplicate when the arguments don't match. If the arguments do match, we still don't need to add the duplicate attribute (the initial one suffices).

Yep, I've made that change locally based on your other comment.

>return nullptr will be out side if (DeclAttr->getType() != A.getType()) { statement to avoid adding one.
 :)  Aaron suggested that, and I've made it locally.  Coming in the next patch... eventually.